pyexcel_text/__init__.py

```diff
@@ -112,6 +113,10 @@ def write_array(self, table):
         import tabulate
         if 'single_sheet_in_book' in self.keywords:
             self.keywords.pop('single_sheet_in_book')
+        if not isinstance(table, Matrix):
+            if not isinstance(table, list):
+                table = list(table)
```

Comment:
If yes, you may want to use this line instead:

https://github.com/pyexcel/pyexcel-io/blob/e26234e5bcfb75e3f0d5355e3121c3daf7cbfdad/pyexcel_io/_compact.py#L22

if not and if not , read a bit awkward.